bundles/org.openhab.binding.luftdateninfo/src/main/java/org/openhab/binding/luftdateninfo/internal/handler/BaseSensorHandler.java

```diff
@@ -0,0 +1,187 @@
+    protected static final HTTPHandler HTTP = new HTTPHandler();
+    private static final LuftdatenInfoConfiguration DEFAULT_CONFIG = new LuftdatenInfoConfiguration();
+    protected final Logger logger = LoggerFactory.getLogger(BaseSensorHandler.class);
+
+    protected LuftdatenInfoConfiguration config = DEFAULT_CONFIG;
+    protected @Nullable ScheduledFuture<?> refreshJob;
+
+    protected ConfigStatus configStatus = ConfigStatus.UNKNOWN;
+
+    public enum ConfigStatus {
+        OK,
+        IS_NULL,
+        SENSOR_IS_NULL,
+        SENSOR_NOT_A_NUMBER,
+        UNKNOWN
+    };
+
+    protected UpdateStatus updateStatus = UpdateStatus.UNKNOWN;
+
+    public enum UpdateStatus {
+        OK,
+        CONNECTION_ERROR,
+        VALUE_ERROR,
+        VALUE_EMPTY,
+        UNKNOWN
+    }
+
+    protected LifecycleStatus lifecycleStatus = LifecycleStatus.UNKNOWN;
+
+    public enum LifecycleStatus {
+        UNKNOWN,
+        RUNNING,
+        INITIALIZING,
+        DISPOSED
+    }
+
+    public BaseSensorHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateFromCache();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        lifecycleStatus = LifecycleStatus.INITIALIZING;
+        scheduler.execute(() -> {
+            startUp();
+        });
+    }
+
+    private void startUp() {
+        config = getConfigAs(LuftdatenInfoConfiguration.class);
+        configStatus = checkConfig(config);
+        if (configStatus == ConfigStatus.OK) {
+            update();
+            if (updateStatus == UpdateStatus.OK) {
+                updateStatus(ThingStatus.ONLINE);
+                logger.debug("Start refresh job at interval {} min.", REFRESH_INTERVAL_MIN);
+                startSchedule();
+            } else {
+                switch (updateStatus) {
+                    case CONNECTION_ERROR:
+                        logger.warn("Update failed due to Connection error. Trying to recover in next refresh");
+                        // start job even if first update isn't valid
+                        startSchedule();
+                        updateStatus(ThingStatus.OFFLINE);
+                        break;
+                    case VALUE_EMPTY:
+                        logger.warn("No values deliverd by Sensor.  Trying to recover in next refresh");
+                        // start job even if first update isn't valid
+                        startSchedule();
+                        updateStatus(ThingStatus.ONLINE);
+                        break;
+                    case VALUE_ERROR:
+                        logger.warn(
+                                "Sensor values doesn't match - please check if Sensor ID is delivering the correct Thing channel values");
+                        updateStatus(ThingStatus.OFFLINE);
```

Comment:
You could feed the message into `updateStatus()` to display it in PaperUI. Then, the log message can be removed, since the framework is logging it. Same for the others.

Reply:
Good hint - I'll do so!